src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -36,21 +37,49 @@ def create_configstore(client,
                        tags=None,
                        assign_identity=None,
                        enable_public_network=None,
-                       disable_local_auth=None):
+                       disable_local_auth=None,
+                       retention_days=None,
+                       enable_purge_protection=None):
     if assign_identity is not None and not assign_identity:
         assign_identity = [SYSTEM_ASSIGNED_IDENTITY]
 
     public_network_access = None
     if enable_public_network is not None:
         public_network_access = 'Enabled' if enable_public_network else 'Disabled'
 
+    if sku.lower() == 'free' and (enable_purge_protection or retention_days):
+        logger.warning("Options '--enable-purge-protection' and '--retention-days' will be ignored when creating a free store.")
+        retention_days = None
+        enable_purge_protection = None
+    elif sku.lower() == 'standard' and not retention_days:
+        retention_days = 7
```

Comment:
The `elif` check can be removed. If no retention days are passed, we dont need to default to 7. Let RP choose the default value.